File: src/test/java/com/github/kayjam/tests/containers/classes/ClassExtendsContainerTest.java
package com.github.kayjam.tests.containers.classes;

import com.github.kayjam.core.Expression;
import com.github.kayjam.core.KayJamLexer;
import com.github.kayjam.core.KayJamParser;
import com.github.kayjam.core.containers.ClassContainer;
import org.junit.BeforeClass;
import org.junit.Test;

import static org.junit.Assert.*;

public class ClassExtendsContainerTest {

    private static KayJamParser parser;

    @BeforeClass
    public static void prepare(){
        parser = new KayJamParser(new KayJamLexer("class Test: ABC {var test = 123;fun test(){return test}}"));
    }

    @Test
    public void test() throws Exception {
        Expression expression = parser.readExpression();

        assertNotNull(expression);
        assertSame(ClassContainer.class, expression.getClass());

        ClassContainer classContainer = (ClassContainer) expression;
        assertEquals("Test", classContainer.name);
        assertEquals("ABC", classContainer.extendsClass);
        assertEquals(0, classContainer.implementsClass.size());
        assertEquals(1, classContainer.children.size());
        assertEquals(1, classContainer.functions.size());
    }
}

File: src/test/scala/com/github/kayjamlang/tests/containers/classes/ClassExtendsContainerTest.scala
package com.github.kayjamlang.tests.containers.classes

import com.github.kayjamlang.core.{KayJamLexer, KayJamParser}
import com.github.kayjamlang.core.containers.ClassContainer
import org.junit.Assert._
import org.junit.{BeforeClass, Test}


object ClassExtendsContainerTest {
  private var parser: KayJamParser = null

  @BeforeClass def prepare(): Unit = {
    parser = new KayJamParser(new KayJamLexer("class Test: ABC {var test = 123;fun test(){return test}}"))
  }
}

class ClassExtendsContainerTest {
  @Test
  @throws[Exception]
  def test(): Unit = {
    val expression = ClassExtendsContainerTest.parser.readTopExpression
    assertNotNull(expression)
    assertSame(classOf[ClassContainer], expression.getClass)
    val classContainer = expression.asInstanceOf[ClassContainer]
    assertEquals("Test", classContainer.name)
    assertEquals("ABC", classContainer.extendsClass)
    assertEquals(0, classContainer.implementsClass.size)
    assertEquals(1, classContainer.variables.size)
    assertEquals(1, classContainer.functions.size)
  }
}